client/boot/index.js

```diff
@@ -235,6 +236,8 @@ function boot() {
 		// Bump general stat tracking overall Newdash usage
 		analytics.mc.bumpStat( { newdash_pageviews: 'route' } );
 
+		context.store.dispatch( resetPageState() );
```

Comment:
Probably would be cool to have some more generic action that other reducers tap into.
in fact, i did something like this here:
https://github.com/Automattic/wp-calypso/commit/3c2612f4ae5f46d128d216f9c9fd6af7ff9ade22#diff-1434133b7ef90d16331d0125afb1fadcR261